Themis/Models/Repository.swift

```diff
@@ -132,6 +135,53 @@ class Node: Hashable, ObservableObject {
     func hash(into hasher: inout Hasher) {
         hasher.combine(path)
     }
+
+    @MainActor
+    public func calculateDiff(templateParticipationId: Int) async {
+        // only calculate once
+        if diffAdded != nil && diffRemoved != nil {
+            return
+        }
+        guard let code = self.code else {
+            return
+        }
+        if templateCode == nil {
+            await fetchTemplateCode(templateParticipationId: templateParticipationId)
```

Comment:
If `calculateDiff` is called twice in quick succession, this will fetch the template code twice, right? The "only calculate once" condition above will not yet be true, so this will be called twice.